src/main/java/reactor/rabbitmq/Receiver.java

```diff
@@ -209,16 +230,27 @@ protected void completeOnChannelShutdown(Channel channel, FluxSink<?> emitter) {
     // TODO consume with dynamic QoS and/or batch ack
 
     public void close() {
-        if (hasConnection.getAndSet(false)) {
-            try {
-                // FIXME use timeout on block (should be a parameter of the Receiver)
-                connectionMono.block().close();
-            } catch (IOException e) {
-                throw new RabbitFluxException(e);
+        if (closingOrClosed.compareAndSet(false, true)) {
+            if (hasConnection.getAndSet(false)) {
+                safelyExecute(
+                        () -> connectionMono.block().close(this.connectionClosingTimeout),
+                        "Error while closing receiver connection"
+                );
+            }
+            if (privateConnectionSubscriptionScheduler) {
+                safelyExecute(
+                        () -> this.connectionSubscriptionScheduler.dispose(),
+                        "Error while disposing connection subscriber scheduler"
+                );
             }
         }
-        if (privateConnectionSubscriptionScheduler) {
-            this.connectionSubscriptionScheduler.dispose();
+    }
+
+    private void safelyExecute(Utils.ExceptionRunnable action, String message) {
+        try {
+            action.run();
+        } catch (Exception e) {
+            LOGGER.warn(message + ": {}", e.getCause());
```

Comment:
Suggest changing this to:

```java
LOGGER.warn(message, e.getCause())
```

I believe that as-is, this will only print `exception.getCause().toString()`, rather than the stacktrace.

Reply:
I was aiming for `LOGGER.warn(message + ": {}", e.getMessage());` actually. I don't want to log a whole stack trace, hopefully this should be enough.

Reply:
Ok, fair enough.

I commented on this, because here, we always log general exceptions with their stacktrace, especially for rare exceptions.  Missing stacktraces have bitten us too many times in prod.

I can't enforce this policy everywhere though.  ;)